### PR TITLE
Add new heuristic for handling OCAMLLIB

### PR DESCRIPTION
**Pull request by dra27**
This is a cut at implementing the idea in #69. I've lazily done it dynamically on src/ocamlbuild_where.ml rather than working out a boolean flag at build time, but I don't think this matters.

### CODE REVIEW

File: src/ocamlbuild_where.ml [hunk 1]
```diff
@@ -11,10 +11,27 @@
...
+  let root, suffix =
+    let ocaml_lib_len = String.length O.ocaml_libdir + 1 in
+    let lib_len = String.length O.libdir_abs in
+    if lib_len < ocaml_lib_len then
+      O.libdir, "ocamlbuild"
+    else
+      let prefix =
+        String.sub O.libdir_abs 0 ocaml_lib_len
+      in
+      if prefix = O.ocaml_libdir ^ "/" then
```
Comment (gasche) [inline src/ocamlbuild_where.ml]: Wouldn't it be more robust to use `O.ocaml_libdir ^ Filename.dir_sep`, and compute the length of this for the `String.sub` call?

Comment (gasche) [inline src/ocamlbuild_where.ml]: Also: I'd rather have the short case first and the long case as an `else` (so `prefix <> ...` I guess).

File: src/ocamlbuild_where.ml [hunk 2]
```diff
@@ -11,10 +11,27 @@
...
+        try
+          let root =
```
Comment (gasche) [inline src/ocamlbuild_where.ml]: calling this `root` is a bit confusing. `subroot`?

File: src/ocamlbuild_where.ml (hunk 1 above)
Comment (dra27) [inline src/ocamlbuild_where.ml]: Blimey, did I hard-code a forward slash... using Ubuntu for a day is clearly some kind of disease :blush:

**Comment by dra27** [comment]
It turns out I was using `"/"` rather than `Filename.dir_sep` for a reason - I have therefore commented it! Code updated to rename root to subroot as suggested and invert the branches (in fact, I merged the two `if`s)

File: src/ocamlbuild_where.ml [hunk 3]
```diff
@@ -11,10 +11,40 @@
...
+      try
+        let normalise_slashes =
+          if Sys.os_type = "Win32" then
+            String.map (function '/' -> '\\' | c -> c)
+          else
+            function s -> s
+        in
+        let subroot =
+          String.sub O.libdir_abs ocaml_lib_len (lib_len - ocaml_lib_len) |>
+          normalise_slashes
```
Comment (gasche) [inline src/ocamlbuild_where.ml]: My personal taste (well, of course I believe it is the only reasonable choise) is to have the infix operator at the start of the next line, `|> normalize_slashes`, but in any case I am happy with code readability now.

Comment (dra27) [inline src/ocamlbuild_where.ml]: Yeah :blush:

**Comment by dra27** [comment]
Pipe tidied up and Changes entry added

**Pull request merged**